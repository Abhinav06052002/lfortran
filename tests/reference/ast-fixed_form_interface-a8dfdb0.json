{
    "basename": "ast-fixed_form_interface-a8dfdb0",
    "cmd": "lfortran --fixed-form --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/fixed_form_interface.f",
    "infile_hash": "d82b348db147a93fac2457e4d9840a31cde6df9b8a576ee6eea3841e",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-fixed_form_interface-a8dfdb0.stdout",
    "stdout_hash": "d35e3b4d3f16894f177dc933d8a7663c7dcf2be6250498ae052f4277",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}