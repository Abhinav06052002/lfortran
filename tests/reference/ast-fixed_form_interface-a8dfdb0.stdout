(TranslationUnit
    [(Program
        fixed_form_interface
        ()
        []
        [(ImplicitNone
            []
            ()
        )]
        [(Declaration
            (AttrType
                TypeInteger
                []
                ()
                ()
                None
            )
            []
            [(i
            []
            []
            ()
            ()
            None
            ())
            (arr
            [(1
            5
            DimensionExpr)]
            []
            ()
            ()
            None
            ())]
            ()
        )
        (Declaration
            (AttrType
                TypeInteger
                []
                ()
                ()
                None
            )
            []
            [(output_unit
            []
            []
            ()
            ()
            None
            ())]
            ()
        )
        (Interface
            (InterfaceHeader)
            ()
            [(InterfaceProc
                (Subroutine
                    my_subroutine
                    [(a)]
                    []
                    ()
                    ()
                    []
                    []
                    []
                    [(Declaration
                        (AttrType
                            TypeInteger
                            []
                            ()
                            ()
                            None
                        )
                        [(AttrIntent
                            In
                        )]
                        [(a
                        []
                        []
                        ()
                        ()
                        None
                        ())]
                        ()
                    )]
                    []
                    []
                    []
                )
            )
            (InterfaceProc
                (Function
                    my_function
                    [(b)]
                    []
                    res
                    ()
                    ()
                    []
                    []
                    []
                    [(Declaration
                        (AttrType
                            TypeInteger
                            []
                            ()
                            ()
                            None
                        )
                        [(AttrIntent
                            In
                        )]
                        [(b
                        []
                        []
                        ()
                        ()
                        None
                        ())]
                        ()
                    )
                    (Declaration
                        (AttrType
                            TypeInteger
                            []
                            ()
                            ()
                            None
                        )
                        []
                        [(res
                        []
                        []
                        ()
                        ()
                        None
                        ())]
                        ()
                    )]
                    []
                    []
                    []
                )
            )]
        )
        (Declaration
            (AttrType
                TypeInteger
                []
                ()
                ()
                None
            )
            []
            [(num
            []
            []
            ()
            2
            Equal
            ())]
            ()
        )
        (Declaration
            (AttrType
                TypeInteger
                []
                ()
                ()
                None
            )
            [(AttrDimension
                [(()
                ()
                DimensionExpr)]
            )
            (SimpleAttribute
                AttrAllocatable
            )]
            [(factors
            []
            []
            ()
            ()
            None
            ())]
            ()
        )]
        [(Allocate
            0
            [(()
            (FuncCallOrArray
                factors
                []
                [(()
                num
                ()
                0)]
                []
                []
                []
            )
            ()
            0)]
            []
            ()
        )
        (Assignment
            0
            factors
            (ArrayInitializer
                ()
                ()
                [3
                4]
            )
            ()
        )
        (Deallocate
            0
            [(()
            factors
            ()
            0)]
            []
            ()
        )
        (Format
            1
            "tr1,a"
            ()
        )
        (Print
            0
            1
            [factors]
            ()
        )
        (DoConcurrentLoop
            0
            ()
            [(ConcurrentControl
                i
                1
                10
                2
            )]
            ()
            []
            [(Assignment
                0
                (FuncCallOrArray
                    arr
                    []
                    [(()
                    i
                    ()
                    0)]
                    []
                    []
                    []
                )
                i
                ()
            )]
            ()
            ()
        )
        (Flush
            0
            [output_unit]
            []
            ()
        )]
        []
    )]
)
